src/Dialogs/Weather/SkysightDialog.cpp

```diff
@@ -82,6 +82,37 @@ class SkysightListItemRenderer
   std::shared_ptr<Skysight> skysight;
 };
 
+//TODO: Could extend this to extract Skysight data at point and display on list (see NOAAListRenderer::Draw(2)
+void
+SkysightListItemRenderer::Draw(Canvas &canvas, const PixelRect rc, unsigned i) {
+  const ComputerSettings &settings = CommonInterface::GetComputerSettings();
+  SkysightActiveMetric m = SkysightActiveMetric(skysight->GetActiveMetric(i));
+
+  tstring first_row = tstring(m.metric->name);
```

Comment:
This always allocates memory and copies the string, even if nothing gets appended, for each repaint for each row. I don't like this kind of bloat.

Reply:
Could you confirm that your point is first_row? I.e. allocate first_row and copy m.metric->name to first_copy, correct?